net/netcheck/netcheck.go

```diff
@@ -731,6 +731,9 @@ func (o *GetReportOpts) getLastDERPActivity(region int) time.Time {
 }
 
 // GetReport gets a report. The 'opts' argument is optional and can be nil.
+// Callers are discouraged from passing a ctx with an arbitrary deadline as this
+// may cause GetReport to return prematurely before all reporting methods have
+// been executed.
```

Comment:
then you need to document the upper bound time this will take before returning?

Reply:
Updated in https://github.com/tailscale/tailscale/pull/13395/commits/9c6e4400e84acbbd82c852e2db769dcd3e07fe13.